pkg/storage/SQL/model_plan/get_tracking_date_by_id_LOADER.sql

```diff
@@ -0,0 +1,63 @@
+WITH QUERIED_IDS AS (
+    -- Translate the input JSON array to a table
+    SELECT id
+    FROM
+        JSON_TO_RECORDSET(:paramTableJSON)
+        AS x(id UUID) --noqa
+),
+
+LATEST_DATES AS (
+    SELECT
+        operational_need.model_plan_id,
+        GREATEST(MAX(operational_need.modified_dts), MAX(operational_need.created_dts)) AS latest_date
+    FROM operational_need
+    WHERE operational_need.model_plan_id IN (SELECT id FROM QUERIED_IDS)
+    GROUP BY operational_need.model_plan_id
+
+    UNION ALL
+
+    SELECT
+        operational_need.model_plan_id,
+        GREATEST(MAX(operational_solution.modified_dts), MAX(operational_solution.created_dts)) AS latest_date
+    FROM operational_solution
+    INNER JOIN operational_need ON operational_need.id = operational_solution.operational_need_id
+    WHERE operational_need.model_plan_id IN (SELECT id FROM QUERIED_IDS)
+    GROUP BY operational_need.model_plan_id
+
+    UNION ALL
+
+    SELECT
+        operational_need.model_plan_id,
+        GREATEST(MAX(operational_solution_subtask.modified_dts), MAX(operational_solution_subtask.created_dts)) AS latest_date
+    FROM operational_solution_subtask
+    INNER JOIN operational_solution ON operational_solution.id = operational_solution_subtask.solution_id
+    INNER JOIN operational_need ON operational_need.id = operational_solution.operational_need_id
+    WHERE operational_need.model_plan_id IN (SELECT id FROM QUERIED_IDS)
+    GROUP BY operational_need.model_plan_id
+
+    UNION ALL
+
+    SELECT
+        operational_need.model_plan_id,
+        GREATEST(MAX(plan_document_solution_link.modified_dts), MAX(plan_document_solution_link.created_dts)) AS latest_date
+    FROM plan_document_solution_link
+    INNER JOIN operational_solution ON operational_solution.id = plan_document_solution_link.solution_id
+    INNER JOIN operational_need ON operational_need.id = operational_solution.operational_need_id
+    WHERE operational_need.model_plan_id IN (SELECT id FROM QUERIED_IDS)
+    GROUP BY operational_need.model_plan_id
+
+    UNION ALL
+
+    SELECT
+        plan_document.model_plan_id,
```

Comment:
I think this is almost correct, but not quite. 

I think we only care about document dates that correspond to a linked document. Maybe it would make sense to add to the select above it instead? (Just add one more join to the document from the document solution link, and find the greatest link or linked document date?)